src/form/input/Input.tsx

```diff
@@ -83,11 +83,12 @@ function Input(props: InputProps) {
     setNumberSeparatorsForLocale
   ] = useState(() => getNumberSeparators(locale));
   const isNumberInput = type === "number";
-  const inputContainerClassName = classNames("input-container", customClassName);
+  const inputContainerClassName = classNames("input-container", customClassName, {
+    "input-container--type-number": isNumberInput
+  });
```

Comment:
```suggestion
  const inputContainerClassName = classNames("input-container", customClassName, `input-container--type-${type}`);
```

I think this is better and this way we would have added the modifier classname for all `type` props